### PR TITLE
drivers/syslog: Implement RTT based log channel

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- drivers/syslog: Implement RTT based log channel
- board/sim: Rename sysview config to segger
- tools/nxstyle: add Segger RTT prototype name into white list

## Impact
New syslog driver

## Testing
Pass CI

### CODE REVIEW

File: include/nuttx/syslog/syslog_rtt.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{
+#else
+#define EXTERN extern
+#endif
+
+#ifdef CONFIG_SYSLOG_RTT
+int syslog_rtt_putc(FAR struct syslog_channel_s *channel, int ch);
+ssize_t syslog_rtt_write(FAR struct syslog_channel_s *channel,
+                         FAR const char *buffer, size_t buflen);
+#endif
```
Comment (pkarashchenko) [inline include/nuttx/syslog/syslog_rtt.h:47]: Minor. We can omit `#ifdef CONFIG_SYSLOG_RTT` wrapper

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Comment by pkarashchenko** [comment]
@xiaoxiang781216 please rebase your changes

**Comment by acassis** [comment]
Please resolve the conflicting file: boards/sim/sim/sim/configs/sysview/defconfig

**Comment by xiaoxiang781216** [comment]
Done.

**Pull request merged**